### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by RicardoEgas**
**In this branch, we implemented the following:**

- tests for the operate and calculate files;
- Installed and used the React Testing Library;
- Used Jest snapshots for all the components;
- Simulated user interaction ;

### CODE REVIEW

File: src/components/__tests__/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    const handleEventMock = jest.fn();
+
+    const tree = renderer
+      .create(<Buttons number="1" handleEvent={handleEventMock} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  // Add more test cases as needed
+});
```
Comment (Gambit142) [inline src/components/__tests__/Button.test.js]: - [ ] Great work using Jest snapshots and the `react-test-renderer` package to test components. In addition to these tests, kindly use **_React Testing Library_** to simulate user interaction for your application and components.👍. You can learn how to use **_React Testing Library_** by reading these articles:
- https://testing-library.com/docs/react-testing-library/intro/
- https://www.robinwieruch.de/react-testing-library/

### Project Requirement

image

File: src/components/__tests__/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import Calc from '../Calculator';
+
+describe('Home component', () => {
+  it('renders correctly', () => {
+    const tree = renderer
+      .create(<Calc />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (Gambit142) [inline src/components/__tests__/Calculator.test.js]: - [ ] Kindly write unit tests for all functionalities and scenarios in the operate.js file and the calculate.js file. 👍 When you write unit tests, bugs can be found easily and quicker. Unit tests help developers detect problems immediately, and then fix them quickly. With fewer resources spent finding bugs, teams can move on to the next phase of a project. Kindly implement this change. 💻 🖱️

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @RicardoEgas @Kccendana 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Use of Jest testing library ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @RicardoEgas 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**